c7n/resources/cloudfront.py

```diff
@@ -181,9 +179,64 @@ def process(self, resources, event=None):
         return results
 
 
+@Distribution.filter_registry.register('mismatch-s3-owner')
+class MismatchS3Owner(Filter):
+    """Check for existence of S3 bucket referenced by Cloudfront,
+       and verify whether owner is different from Cloudfront account owner.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: mismatch-s3-owner
+                resource: distribution
+                filters:
+                  - type: mismatch-s3-owner
+   """
+
+    schema = type_schema(
+        'mismatch-s3-owner')
+
+    permissions = ('s3:ListBuckets',)
+    retry = staticmethod(get_retry(('Throttling',)))
+
+    def is_s3_domain(self, domain_name):
+        if domain_name.endswith('.s3.amazonaws.com'):
+            return True
+        else:
+            return False
+
+    def get_s3_domain(self, domain_name):
+        if domain_name.endswith('.s3.amazonaws.com'):
```

Comment:
why not use the function you defined earlier? `is_s3_domain`?

Reply:
Similar to my approach with `is_s3_domain`, I wanted to make it extensible with additional URL strings if that's necessary later on.

Reply:
in that case you would be violating DRY as you would redefine url strings in both locations, get_s3_domain and is_s3_domain, extensibility should exist within a single function as to ensure consistency in behavior

Reply:
Fair enough, I can roll those into a single function.